zkir/Dialect/Field/Conversions/FieldToModArith/FieldToModArith.cpp

```diff
@@ -506,6 +685,59 @@ struct ConvertSquare : public OpConversionPattern<SquareOp> {
       rewriter.replaceOp(op, f2);
       return success();
     }
+    if (auto f3Type = dyn_cast<CubicExtFieldType>(fieldType)) {
+      // construct xi as a mod arith constant
```

Comment:
Please use https://github.com/kroma-network/tachyon/blob/e7b13063c6f110b9851655cca740ee62ffb41137/tachyon/math/finite_fields/cubic_extension_field.h#L370-L408

instead of implementing square op here, what about delegating logic to a dedicated class, something like `CubicExtensionField`? similar to how we do in `MontReducer`!